### PR TITLE
document current compiler support

### PR DESCRIPTION
**Pull request by marehr**
Solves #1774 (document current compiler support)

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -26,6 +26,8 @@ Linux, Mac OSX, FreeBSD:
   * LLVM / Clang ≥ 3.5
 Windows:
   * Visual C++ ≥ 14.0 / Visual Studio ≥ 2015
+  * Intel Compiler ≥ 16.0.3
```
Comment (h-2) [inline README.rst]: * Intel Compiler ≥ 16.0.3 / Visual Studio ≥ 2015u2

File: README.rst [hunk 2]
```diff
@@ -26,6 +26,8 @@ Linux, Mac OSX, FreeBSD:
...
+  * Clang/C2 ≥ 3.8.0 (experimental and requires CMake ≥ 3.6)
```
Comment (h-2) [inline README.rst]: * Clang/C2 ≥ 3.8.0 / Visual Studio ≥ 2015u3 [experimental, requires CMake ≥ 3.6]

**Comment by h-2** [comment]
Can you also update the CHANGELOG?

**Comment by marehr** [comment]
I thought someone else does the changelog

**Comment by h-2** [comment]
> I thought someone else does the changelog

No, we decided that it's best if every (important) PR documents its own changes in the changelog. It is more consistent and ensures that nothing gets lost.

**Comment by marehr** [comment]
So you want me to say, that nothing important changed until now?

**Comment by marehr** [comment]
Ups, I looked into the master branch.

**Comment by h-2** [comment]
> So you want me to say, that nothing important changed until now?

:pensive:

Please add something like 
```
Infrastructure Updates

Build System:

* The Intel Compiler is now fully supported on Linux and Windows, both 32bit and 64bit; it builds faster binaries and supports some functionality not available in MSVC
* On Windows there is now experimental support for Clang/C2, the Microsoft version of the clang compiler. 
* Please see the manual [link] for more information on how to use these compilers.
```

**Comment by marehr** [comment]
Forgot to say it, but done.

**Comment by h-2** [comment]
One more small thing: Can you add two sentences or a screenshot to the manual that explain how to install Clang/C2 on Visual Studio? Thanks!

**Comment by marehr** [comment]
I already linked to https://blogs.msdn.microsoft.com/vcblog/2015/12/04/clang-with-microsoft-codegen-in-vs-2015-update-1.

**Pull request merged**